### ISSUE TITLE
Fix the background loading order

### ISSUE THREAD

**Issue by tederis** [urho3d/Urho3D#2201]
I faced with a problem when I try to load a complex resource with several sub-resources via BackgroundLoadResource(), and BackgroundLoader calls the EndLoad() on caller resource before their dependencies are finished. In example, Textures and Technique resources will be loaded and cached only after than Material do this. As a result, at the time of the EndLoad() called, the sub-resources are not yet ready and will be loaded in the main thread.

**Comment by Lumak** [comment]
Nice that you were able to trace and pin point the problem.  What changes have you made to make your loading work?

**Comment by tederis** [comment]
I tried to bring some changes to the code of Urho and move processing dependencies code part(lines 93-104) from background thread to the FinishBackgroundLoading() method. 
And only after that, background loading started to work properly, but isn't the solution because in this case sync(via GetResource()) loading will be broken.

**Comment by Lumak** [comment]
Thanks for the feedback.
@cadaver mentioned changing the loading order in another issue tracker, but I can't remember what that order was or the issue tracker.
